.github/workflows/test.yml

```diff
@@ -42,3 +40,48 @@ jobs:
 
     - name: run unit tests (includes lint)
       run: yarn test
+
+
+  node-integration-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['18.x', '20.x']
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: yarn install
+      run: yarn
+
+    - name: run node iTests
+      env:
+        GEMINI_API_KEY: ${{secrets.GEMINI_API_KEY}}
+      run: |
+        cd packages/main
```

Comment:
`yarn --cwd packages/main test:node:integration` if you want to save a line

Reply:
Done!